app/src/main/java/com/epfl/drawyourpath/utils/Utils.kt

```diff
@@ -264,4 +269,91 @@ object Utils {
         val y = ln(tan(lat) + (1 / cos(lat)))
         return Point.create(long.toFloat(), y.toFloat())
     }
+
+    /**
+     * Helper function to download an get the ML model
+     * @return the ML model as a future of DigitalInkRecognitionModel
+     */
+    fun downloadModelML(): CompletableFuture<DigitalInkRecognitionModel> {
+        var result = CompletableFuture<DigitalInkRecognitionModel>()
+        var modelIdentifier: DigitalInkRecognitionModelIdentifier? = null
+        try {
+            modelIdentifier = DigitalInkRecognitionModelIdentifier.fromLanguageTag("zxx-Zsym-x-autodraw")
+        } catch (e: MlKitException) {
+            result.completeExceptionally(Error("The ML model identifier language tag failed to parse."))
+            return result
+        }
+        if (modelIdentifier == null) {
+            result.completeExceptionally(Error("The ML model was not found."))
+            return result
+        }
+
+        var model = DigitalInkRecognitionModel.builder(modelIdentifier!!).build()
+        val remoteModelManager = RemoteModelManager.getInstance()
+        remoteModelManager.isModelDownloaded(model)
+            .addOnSuccessListener() { is_downloaded: Boolean ->
+                if (!is_downloaded) {
+                    remoteModelManager.download(model, DownloadConditions.Builder().build())
+                        .addOnSuccessListener {
+                            Log.i("Utils", "Model downloaded")
+                            result.complete(model)
+                        }
+                        .addOnFailureListener { e: Exception ->
+                            result.completeExceptionally(Error("Fail to download the model."))
```

Comment:
Same remark for exception message, might be helpful to show the reason the download failed.